### PR TITLE
Update to newest eslint-config-edx. Remove IIFEs in Node code.

### PR DESCRIPTION
**Pull request by bjacobel**
Update to the version of eslint-config-edx that contains fixes to 'use strict'; rule. Node code doesn't need to be wrapped in IIFEs and should 'use strict'; at global scope.

(This undoes all the wrapping of this code I thought I needed to do a few weeks ago....)

- [ ] @andy-armstrong 
- [x] @alisan617

### CODE REVIEW

File: gulp/tasks/clean.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
+/* eslint-env node */
```
Comment (andy-armstrong) [inline gulp/tasks/clean.js]: I thought you had a rule so that the gulp directory was automatically treated as being a node environment?

**Comment by andy-armstrong** [comment]
👍

File: gulp/tasks/clean.js (hunk 1 above)
Comment (bjacobel) [inline gulp/tasks/clean.js]: I thought I could just do 

`./gulp/.eslintrc.json`
```
{
  "env": {
    "node: true
  }
}
```

but that wasn't working for some reason. I'll take another look at it before I merge this. Got sidetracked when my jenkins build finished :)

Comment (bjacobel) [inline gulp/tasks/clean.js]: Updated to fix this, the folder-level .eslintrc.json now has the right configuration to make this folder pick up the right setting for the `strict` rule.

**Comment by bjacobel** [comment]
Ready for re-review - @alisan617, if you could when you get a chance? Thanks!

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
     "del": "~2.2.0",
     "edx-pattern-library": "~0.12.5",
```
Comment (alisan617) [inline package.json]: any reason why UXPL is not updated to the latest?

Comment (bjacobel) [inline package.json]: We generally don't update deps until we need to. Nothing visual has changed in the UITK for a while.

**Pull request merged**